components/series/stackedBar.js

```diff
@@ -6,77 +6,39 @@
             var container;
 
-            // takes an object with values associated with each property, and
-            // converts it into an array of values. Each value has the xValue associated
-            // with it.
-            //
-            // For example, this object:
-            //
-            // obj = { county: 'North Tyneside', labour: 23, conservative: 55 }
-            //
-            // becomes this ...
-            //
-            // [
-            //   { xValue: 'North Tyneside', name: 'labour', previousValue: 0, currentValue: 23},
-            //   { xValue: 'North Tyneside', name: 'conservative', previousValue: 23, currentValue: 78},
-            // ]
-            function objectDatapointToArray(obj) {
-                var values = [];
-                var yTotal = 0;
-                var xVal = obj[stackedBar.xValueKey.value];
-                for (var propertyName in obj) {
-                    if (obj.hasOwnProperty(propertyName) && propertyName !== stackedBar.xValueKey.value) {
-                        var previous = yTotal;
-                        yTotal += Number(obj[propertyName]);
-                        values.push({
-                            'name': propertyName,
-                            'previousValue': previous,
-                            'currentValue': yTotal,
-                            'xValue': xVal
-                        });
-                    }
-                }
-                return values;
-            }
-
             selection.each(function(data) {
 
                 // add a 'root' g element on the first enter selection. This ensures
                 // that it is just added once
                 container = d3.select(this);
 
-                var keyFunction = function(d) {
-                    return d[stackedBar.xValueKey.value];
-                };
-                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', data, keyFunction);
+                // If the user has already set stacked data then we don't need to re-compute it.
+                var layers = stackedBar.stackedData.value;
+                if (typeof layers === 'undefined' || layers.length === 0) {
+                    layers = stackedBar.layout.value(data);
+                }
+
+                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', layers);
 
+                stackedBar.decorate.value(g);
 
-                // create a join for each bar
                 var bar = g.selectAll('rect')
-                    .data(function(d) { return objectDatapointToArray(d); })
+                    .data(function(d) { return d; })
                     .enter()
                     .append('rect');
 
-                // compute the bar width from the x values
-                var xValues = data.map(function(d) {
-                    return stackedBar.xScale.value(d[stackedBar.xValueKey.value]);
-                });
+                // Compute the bar width from the x values
```

Comment:
Yes, it would. We should be using the accessor functions defined the stacked layout, as it has accessor functions defined on it for the `x` & `y` values. However, it doesn't define an accessor function for `y0`, so that's something we'd need to define on the series itself. 

Since the stack layout also has a setter function for `y0`, it's worth noting that there would be an implicit link between `y0`'s setter and getter and user's would have to remember to update both. This would probably be nicer once I've rebound the methods (as you suggested) as both the getter and the setter would be accessible on the series.